### PR TITLE
Bug 1954087: added test for manage config maps in targetconfigcontroller

### PR DESCRIPTION
**Pull request by atantawi**

### CODE REVIEW

**Review by damemi — COMMENTED** [review]
I think this is exactly what we're looking for in these kind of unit tests @atantawi @wangchen615 . We could probably add similar tests for the other functions as well (ie, `managePod...`, `manageDeployment...`, etc).

@ingvagabund wdyt?

File: pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go [hunk 1]
```diff
@@ -2,30 +2,186 @@ package targetconfigcontroller
...
+			want:    cmNo,
+			want1:   true,
+			wantErr: false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			// need a client for each test
+			tt.args.client = fake.NewSimpleClientset().CoreV1()
+			got, got1, err := manageKubeSchedulerConfigMap_v311_00_to_latest(tt.args.client, tt.args.recorder, tt.args.configSchedulerLister)
```
Comment (ingvagabund) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go:170]: Given `tt.args.client` is used only once in this block, you might pass `fake.NewSimpleClientset().CoreV1()` directly into `manageKubeSchedulerConfigMap_v311_00_to_latest`

File: pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go [hunk 2]
```diff
@@ -2,30 +2,186 @@ package targetconfigcontroller
...
 	"k8s.io/client-go/tools/cache"
-
-	operatorv1 "github.com/openshift/api/operator/v1"
-	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
-	"github.com/openshift/library-go/pkg/operator/events"
 )
 
 var codec = scheme.Codecs.LegacyCodec(scheme.Scheme.PrioritizedVersionsAllGroups()...)
 
+var configLow = &configv1.Scheduler{
```
Comment (ingvagabund) [inline pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go:34]: `configLowNodeUtilization` is longer but better to understand by its name, the same for other occurences of short names:
- `configHigh` -> `configHighNodeUtilization`
- `configNo` -> `configNoScoring`
- `kscLow` and alike
- `cmLow` and alike

**Review by ingvagabund — COMMENTED** [review]
Just nits, otherwise LGTM

**Comment by ingvagabund** [comment]
> We could probably add similar tests for the other functions as well (ie, managePod..., manageDeployment..., etc).

absolutely!!!

**Comment by damemi** [comment]
/retitle Bug 1954087: added test for manage config maps in targetconfigcontroller
@atantawi if you address @ingvagabund's comments we can merge this
/approve

File: pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go (hunk 1 above)
Comment (atantawi) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go]: Addressed in commit 764f14d9c4a0402f6d17b2fa017d13257f6f2bff

File: pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go (hunk 2 above)
Comment (atantawi) [reply pkg/operator/targetconfigcontroller/targetconfigcontroller_test.go]: Addressed in commit 764f14d9c4a0402f6d17b2fa017d13257f6f2bff

**Comment by damemi** [comment]
@atantawi thanks, please squash down to 1 commit
Also, when catching your branch up to master it is fine to just rebase rather than merge (to avoid extraneous merge commits)

**Comment by atantawi** [comment]
> @atantawi thanks, please squash down to 1 commit
> Also, when catching your branch up to master it is fine to just rebase rather than merge (to avoid extraneous merge commits)

Done

**Pull request merged**